interfaces/builtin/unity7.go

```diff
@@ -460,22 +502,65 @@ dbus (send)
 deny /{dev,run,var/run}/shm/lttng-ust-* rw,
 `
 
-// http://bazaar.launchpad.net/~ubuntu-security/ubuntu-core-security/trunk/view/head:/data/seccomp/policygroups/ubuntu-core/16.04/unity7
-const unity7ConnectedPlugSecComp = `
-# Description: Can access Unity7. Restricted because Unity 7 runs on X and
-# requires access to various DBus services and this environment does not prevent
-# eavesdropping or apps interfering with one another.
+const unity7ConnectedPlugSeccomp = `
+# Description: Can access Unity7. Note, Unity 7 runs on X and requires access
+# to various DBus services and this environment does not prevent eavesdropping
+# or apps interfering with one another.
 
 # X
 shutdown
 `
 
-// NewUnity7Interface returns a new "unity7" interface.
-func NewUnity7Interface() interfaces.Interface {
-	return &commonInterface{
-		name: "unity7",
-		connectedPlugAppArmor: unity7ConnectedPlugAppArmor,
-		connectedPlugSecComp:  unity7ConnectedPlugSecComp,
-		reservedForOS:         true,
+type Unity7Interface struct{}
+
+func (iface *Unity7Interface) Name() string {
+	return "unity7"
+}
+
+func (iface *Unity7Interface) String() string {
+	return iface.Name()
+}
+
+func (iface *Unity7Interface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.Plug, slot *interfaces.Slot) error {
+	var pathBuf bytes.Buffer
+	// Unity7 will take the desktop filename and convert all '-' (and '.',
+	// but we don't care about that here because the rule above already
+	// does that) to '_'. Since we know that the desktop filename starts
+	// with the snap name, perform this conversion on the snap name.
+	pathBuf.WriteString(strings.Replace(plug.Snap.Name(), "-", "_", -1))
```

Comment:
I think you can just do `strings.Replace()` here. What is the need for the buffer?